tests/cnf/core/network/sriov/tests/qinq.go

```diff
@@ -544,6 +578,152 @@ var _ = Describe("QinQ", Ordered, Label(tsparams.LabelQinQTestCases), ContinueOn
+			err = netenv.WaitForSriovAndMCPStable(
+				APIClient, tsparams.MCOWaitTimeout, time.Minute, NetConfig.CnfMcpLabel, NetConfig.SriovOperatorNamespace)
+			Expect(err).ToNot(HaveOccurred(), "Failed cluster is not stable")
+
+			By("Creating a new instance of NMstate instance")
+			err = netnmstate.CreateNewNMStateAndWaitUntilItsRunning(7 * time.Minute)
+			Expect(err).ToNot(HaveOccurred(), "Failed to create NMState instance")
+
+			By("Validating SR-IOV interfaces")
+			workerNodeList, err = nodes.List(APIClient,
+				metav1.ListOptions{LabelSelector: labels.Set(NetConfig.WorkerLabelMap).String()})
+			Expect(err).ToNot(HaveOccurred(), "Failed to discover worker nodes")
+
+			Expect(sriovenv.ValidateSriovInterfaces(workerNodeList, 2)).ToNot(HaveOccurred(),
+				"Failed to get required SR-IOV interfaces")
+			sriovInterfacesUnderTest, err = NetConfig.GetSriovInterfaces(2)
+			Expect(err).ToNot(HaveOccurred(), "Failed to retrieve SR-IOV interfaces for testing")
+
+			if sriovenv.IsMellanoxDevice(sriovInterfacesUnderTest[0], workerNodeList[0].Object.Name) {
+				configureSriovMlnxFirmwareOnWorkersAndWaitMCP(workerNodeList, sriovInterfacesUnderTest[0], true, 5)
+			}
+
+			By("Creating SR-IOV VFs via NMState")
+			err = netnmstate.ConfigureVFsAndWaitUntilItsConfigured(
+				configureNMStatePolicyName,
+				sriovInterfacesUnderTest[0],
+				NetConfig.WorkerLabelMap,
+				5,
+				netparam.DefaultTimeout)
+			Expect(err).ToNot(HaveOccurred(), "Failed to create VFs via NMState")
+
+			err = sriovenv.WaitUntilVfsCreated(workerNodeList, sriovInterfacesUnderTest[0], 5, netparam.DefaultTimeout)
+			Expect(err).ToNot(HaveOccurred(), "Expected number of VFs are not created")
+
+			By("Configure SR-IOV with flag ExternallyManaged true")
+			err = sriovenv.CreateSriovNmStateConfiguration(sriovAndResourceNameExManagedTrue, sriovInterfacesUnderTest[0], true)
+			Expect(err).ToNot(HaveOccurred(),
+				fmt.Sprintf("Failed to create sriov configuration via nmstate %s", err))
+
+			By("Define and create sriov-network for the promiscuous client")
+			_, err = sriov.NewNetworkBuilder(APIClient,
+				srIovNetworkPromiscuous, NetConfig.SriovOperatorNamespace, tsparams.TestNamespaceName,
+				sriovAndResourceNameExManagedTrue).WithTrustFlag(true).Create()
+			Expect(err).ToNot(HaveOccurred(),
+				fmt.Sprintf("Failed to create sriov network srIovNetworkPromiscuous %s", err))
+
+			By("Define and create sriov-network with 802.1ad S-VLAN")
+			vlan, err := strconv.Atoi(NetConfig.VLAN)
+			Expect(err).ToNot(HaveOccurred(), "Failed to convert VLAN value")
+			defineAndCreateSrIovNetworkWithQinQ(srIovNetworkDot1AD, sriovAndResourceNameExManagedTrue, dot1ad,
+				uint16(vlan))
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to create sriov network srIovNetworkDot1AD %s", err))
+			By("Verify SR-IOV Device IDs for interface under test")
+			sriovDeviceID = discoverInterfaceUnderTestDeviceID(srIovInterfacesUnderTest[0],
```

Comment:
Can you use existing func sriovenv.IsMellanoxDevice ?

Reply:
Removed